lib/domain/bloc/controller.dart

```diff
@@ -222,13 +297,105 @@ class VideoEditorController extends ChangeNotifier {
     notifyListeners();
   }
 
+  /// Return metadata of the video file
+  Future<void> getMetaData(
+    String filePath, {
+    required Function(Map<dynamic, dynamic>? metadata) onCompleted,
+  }) async {
+    await FFprobeKit.getMediaInformationAsync(filePath,
+        (MediaInformationSession session) async {
+      final information = (session).getMediaInformation();
+      onCompleted.call(information?.getAllProperties());
+    });
+  }
+
+  _setCurrentBitRates() async {
+    Completer completer = Completer();
```

Comment:
This function is a `Future<void>`,is this `Completer` needed ?

Reply:
Yes it has a function that has a callback, which he has to wait for.